commands/update_organization_setcredentials.go

```diff
@@ -0,0 +1,349 @@
+	cmdAzureTenantID       string
+	cmdAzureClientID       string
+	cmdAzureSecretKey      string
+
+	// Here we briefly store the info which provider we are dealing with
+	provider string
+)
+
+type updateOrgSetCredentialsArguments struct {
+	apiEndpoint         string
+	authToken           string
+	scheme              string
+	verbose             bool
+	organizationID      string
+	awsAdminRole        string
+	awsOperatorRole     string
+	azureSubscriptionID string
+	azureTenantID       string
+	azureClientID       string
+	azureSecretKey      string
+}
+
+type updateOrgSetCredentialsResult struct {
+	credentialID string
+}
+
+func init() {
+	UpdateOrgSetCredentialsCommand.Flags().StringVarP(&cmdOrganizationID, "organization", "o", "", "ID of the organization to set credentials for")
+	UpdateOrgSetCredentialsCommand.Flags().StringVarP(&cmdAWSOperatorRoleARN, "aws-operator-role", "", "", "AWS ARN of the role to use for operating clusters")
+	UpdateOrgSetCredentialsCommand.Flags().StringVarP(&cmdAWSAdminRoleARN, "aws-admin-role", "", "", "AWS ARN of the role to be used by Giant Swarm staff")
+	UpdateOrgSetCredentialsCommand.Flags().StringVarP(&cmdAzureSubscriptionID, "azure-subscription-id", "", "", "ID of the Azure subscription to run clusters in")
+	UpdateOrgSetCredentialsCommand.Flags().StringVarP(&cmdAzureTenantID, "azure-tenant-id", "", "", "ID of the Azure tenant to run clusters in")
+	UpdateOrgSetCredentialsCommand.Flags().StringVarP(&cmdAzureClientID, "azure-client-id", "", "", "ID of the Azure service principal to use for operating clusters")
+	UpdateOrgSetCredentialsCommand.Flags().StringVarP(&cmdAzureSecretKey, "azure-secret-key", "", "", "Secret key for the Azure service principal to use for operating clusters")
+
+	UpdateOrganizationCommand.AddCommand(UpdateOrgSetCredentialsCommand)
+}
+
+func defaultUpdateOrgSetCredentialsArguments() updateOrgSetCredentialsArguments {
+	endpoint := config.Config.ChooseEndpoint(cmdAPIEndpoint)
+	token := config.Config.ChooseToken(endpoint, cmdToken)
+	scheme := config.Config.ChooseScheme(endpoint, cmdToken)
+
+	return updateOrgSetCredentialsArguments{
+		apiEndpoint:         endpoint,
+		authToken:           token,
+		scheme:              scheme,
+		organizationID:      cmdOrganizationID,
+		verbose:             cmdVerbose,
+		awsAdminRole:        cmdAWSAdminRoleARN,
+		awsOperatorRole:     cmdAWSOperatorRoleARN,
+		azureClientID:       cmdAzureClientID,
+		azureSecretKey:      cmdAzureSecretKey,
+		azureSubscriptionID: cmdAzureSubscriptionID,
+		azureTenantID:       cmdAzureTenantID,
+	}
+}
+
+func updateOrgSetCredentialsPreRunOutput(cmd *cobra.Command, cmdLineArgs []string) {
+	args := defaultUpdateOrgSetCredentialsArguments()
+	err := verifyUpdateOrgSetCredentialsPreconditions(args)
+
+	if err == nil {
+		return
+	}
+
+	handleCommonErrors(err)
+
+	// From here on we handle errors that can only occur in this command
+	headline := ""
+	subtext := ""
+
+	switch {
+	case err.Error() == "":
```

Comment:
What is this case for ?
Isnt' this already catched by `err == nil` ?

Reply:
Removed